src/systemd.c

```diff
@@ -0,0 +1,286 @@
+	char *object = NULL;
+	sd_bus *bus = NULL;
+	int ret = 0;
+	
+	int childPID = fork();
+
+	if(childPID < 0){ //No child process created
+		fprintf(stderr, "fork() failed: %d:%s\n", childPID, strerror(-childPID));
+	}
+	else if (childPID == 0){ //This is the child process, sleep forever
+//		fprintf(stdout, "Child Process PID: %d\n", getpid());
+		permanent_sleep();
+	}
+	else{ //This is the parent process, move child process to the desired slice/scope
+
+//		fprintf(stdout, "Parent Process PID: %d\n", getpid());
+		
+		ret = sd_bus_open_system(&bus);
+		if (ret<0){
+			fprintf(stderr, "sd_bus_open_system failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_new_method_call(bus, &m, "org.freedesktop.systemd1",
+			"/org/freedesktop/systemd1", "org.freedesktop.systemd1.Manager",
+			 "StartTransientUnit");
+		if (ret<0){
+			fprintf(stderr, "sd_bus_message_new_method_call failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_append(m, "ss", scope_name, mode);
+		if (ret<0){
+			fprintf(stderr, "1st sd_bus_message_append failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_open_container(m, 'a', "(sv)");
+		if (ret<0)
+			{fprintf(stderr, "sd_bus_message_open_container failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_append(m, "(sv)", "PIDs", "au", 1, (uint32_t) childPID);
+		if (ret<0){
+			fprintf(stderr, "2nd sd_bus_message_append failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		if(delegated == 1){
+			ret = sd_bus_message_append(m, "(sv)", "Delegate", "b", 1);
+			if (ret<0){
+				fprintf(stderr, "3rd sd_bus_message_append failed: %d:%s\n", ret, strerror(-ret));
+				goto out;
+			}
+		}
+
+		ret = sd_bus_message_append(m, "(sv)", "Slice", "s", slice_name);
+		if (ret<0){
+			fprintf(stderr, "4th sd_bus_message_append failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_close_container(m);
+		if (ret<0){
+			fprintf(stderr, "sd_bus_message_close failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_append(m, "a(sa(sv))", 0);
+		if (ret<0){
+			fprintf(stderr, "5th sd_bus_message_append failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_call(bus, m, 0, &error, &reply);
+		if (ret<0){
+			fprintf(stderr, "sd_bus_call failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+
+		ret = sd_bus_message_read(reply, "o", &object);
+		if (ret<0){
+			fprintf(stderr, "sd_bus_message_reply failed: %d:%s\n", ret, strerror(-ret));
+			goto out;
+		}
+		
+		// Parent thread will return PID of child thread
+		ret = childPID;
+	}
+
+	//@todo Does object need to be freed?
+	out:
+	if(m != NULL){
+		sd_bus_message_unref(m);
+	}
+	if(reply != NULL){
+		sd_bus_message_unref(reply);
+	}
+	if(bus != NULL){
+		sd_bus_unref(bus);
+	}
+	
+	return ret;
+}
+
+int cgroup_create_scope_user_slice(char *scope_name, int delegated, char *mode){
```

Comment:
Fixed, with present code left in block comment. As I said before, my original attempt to do this had sporadic errors crop up when getxattr was called after the wrapper function, specifically error 1-Operation Invalid (break happened on what is now line 235). I couldn't pin down what caused the error so as to consistently trigger it, but repeated runs revealed it was there, so I went with duplicate code over potentially error-generating code. This hasn't recurred since re-implementing the wrapper, but I figured I'd note it was there at some point in case it comes back.

Reply:
Commented code removed.